community/modules/project/service-account/outputs.tf

```diff
@@ -14,47 +14,20 @@
  * limitations under the License.
  */
 
-output "email" {
-  description = "Service account email (for single use)."
-  value       = module.service_accounts.email
-}
-output "emails" {
-  description = "Service account emails by name."
-  value       = module.service_accounts.emails
-}
-output "emails_list" {
-  description = "Service account emails s list."
-  value       = module.service_accounts.emails_list
-}
 output "iam_email" {
-  description = "IAM-format service account email (for single use)."
+  description = "IAM-format service account email"
   value       = module.service_accounts.iam_email
 }
-output "iam_emails" {
-  description = "IAM-format service account emails by name."
-  value       = module.service_accounts.iam_emails
-}
-output "iam_emails_list" {
-  description = "IAM-format service account emails s list."
-  value       = module.service_accounts.iam_emails_list
-}
+
 output "key" {
-  description = "Service account key (for single use)."
+  description = "Service account key (if creation was requested)"
   value       = module.service_accounts.key
 }
-output "keys" {
-  description = "Map of service account keys."
-  value       = module.service_accounts.keys
-}
+
 output "service_account" {
```

Comment:
Nothing to address now, more like things to consider. 
I think this unnecessary nesting is ugly, it would be great if module outputs were just:
```hcl
.../service-account/outputs.tf
output "email" { ... }
output "scopes" { ... }
```
And module itself could be used as an object
```yaml
- id: sa
  source: .../service-account
  ...
- id: vm
  source: modules/compute/vm-instance
  settings:
    service_account: $(sa)
```
Or simply:
```yaml
- id: vm
  source: modules/compute/vm-instance
  use: [sa]
```

Terraform (1.2) does support it, but it would require some changes in toolkit code.

Reply:
Would this translate in HCL to `service_account = module.sa`?